service/worker/archiver/activities.go

```diff
@@ -86,79 +79,49 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 		NextEventID:          request.NextEventID,
 		CloseFailoverVersion: request.CloseFailoverVersion,
 	}, carchiver.GetHeartbeatArchiveOption(), carchiver.GetNonRetriableErrorOption(errUploadNonRetriable))
+	if err != nil && err.Error() == errUploadNonRetriable.Error() {
+		return cadence.NewCustomError(errUploadNonRetriable.Error())
```

Comment:
Add logging if the call fails? (Sorry, this is my fault, I forget to do the logging.)
Convert to Cadence custom error in defer()?
Why only non-retriable error is converted to custom error?